crates/cargo-bsp/src/server/dispatch.rs

```diff
@@ -98,18 +100,28 @@ impl<'a> RequestDispatcher<'a> {
             params,
             global_state_snapshot,
         );
-        match request_handle {
-            Ok(request_handle) => {
-                self.global_state.handlers.insert(req.id, request_handle);
-            }
-            Err(e) => {
-                let response =
-                    Response::new_err(req.id, ErrorCode::InternalError as i32, e.to_string());
-                self.global_state.respond(response);
-            }
-        }
+        self.update_handlers(request_handle, req)
+    }
 
-        self
+    pub(crate) fn on_cargo_check_run<R>(&mut self) -> &mut Self
+    where
+        R: bsp_types::requests::Request + 'static,
+        R::Params: CreateCommand + ParamsTarget + Send + fmt::Debug,
+        R::Result: Serialize,
+    {
+        let (req, params, _) = match self.parse::<R>() {
+            Some(it) => it,
+            None => return self,
+        };
+        let sender_to_main = self.global_state.handlers_sender.clone();
+        let global_state_snapshot = self.global_state.snapshot();
+        let request_handle = RequestHandle::spawn_check::<R>(
+            Box::new(move |msg| sender_to_main.send(msg).unwrap()),
+            req.id.clone(),
+            params,
+            global_state_snapshot,
+        );
+        self.update_handlers(request_handle, req)
```

Comment:
to jest niemal copy past `on_cargo_run`, nie dałoby się tej logiki jakoś uogólnić?

Reply:
Problemem są tutaj znowu traity jakie implementuje R::Params i R::Result (dokładnie tak jak opisałam przy cargo_actor vs request_actor)

Reply:
Niestety po dyskusji o `enabled_features` z Tomkiem okazało się, że `spawn_check` nie może brać `GlobalStateSnapshot`, a musi `GlobalState`, więc nie będą zgadzały się typy w F.

Reply:
ugh, szkoda